staging/src/k8s.io/apimachinery/pkg/runtime/interfaces.go

```diff
@@ -256,6 +267,22 @@ type Object interface {
 	DeepCopyObject() Object
 }
 
+// CacheableObject allows an object to cache its different serializations
+// to avoid performing the same serialization multiple times.
+type CacheableObject interface {
+	// CacheEncode writes an object to a stream. The <encode> function will
+	// be used in case of cache miss. If CacheableObject is a wrapper, then
+	// the wrapped object should be passed to <encode> function.
+	// CacheEncode assumes that for two different calls with the same <id>,
+	// <encode> function will also be the same.
+	CacheEncode(id Identifier, encode func(Object, io.Writer) error, w io.Writer) error
+	// GetObject returns an object to be encoded.
+	// The object returned by GetObject should be the same as the one that is supposed
+	// to be passed to <encode> function in CacheEncode method.
+	// If CacheableObject is a wrapper, the wrapped object should be returned.
+	GetObject() Object
```

Comment:
see comment at https://github.com/kubernetes/kubernetes/pull/81914#discussion_r322808850, the ownership semantics for this method and the object passed to encode need to be clarified for implementers and consumers of the Encoder and CacheableObject interfaces

Reply:
Clarified that it should be a deep copy and added short explanation for it.